corgi/core/models.py

```diff
@@ -904,75 +910,84 @@ def latest_components(
         self,
         include: bool = True,
     ) -> "ComponentQuerySet":
-        """Return components from latest builds."""
-        names = self.values_list("name", flat=True).order_by().distinct().iterator()
+        """Return components from latest builds across all product streams."""
+        distinct_components = (
+            self.values_list("namespace", "name", "arch").order_by().distinct().iterator()
+        )
```

Comment:
There's no restriction on type, so if I include a name in the filter such as 'cloud.google.com/go/monitor', I get results such as 'pkg:golang/cloud.google.com/go/monitoring@v1.1.0' which uses semantic versioning, not RPM versioning, so the results could be incorrect.

```suggestion
        # This should probably go in constants.py
        RPM_SCHEMANTIC_CONDITION = Q(type="RPM") | Q(type="OCI", namespace="REDHAT")
        distinct_components = (
            self.filter(RPM_SCHEMANTIC_CONDITION).values_list("namespace", "name", "arch").order_by().distinct().iterator()
        )
```